src/js/entities/shapes.js

```diff
@@ -0,0 +1,56 @@
+import {Point} from './points';
+class Shape {
+    constructor(points) {
+        this.path2D = null;
+        this.intersectingShapes = [];
+        this.points = points;
+    }
+
+    draw(context) {
+        const startPoint = this.points[0];
+        this.path2D = new Path2D();
+        this.path2D.moveTo(startPoint.x, startPoint.y);
+        this.points.forEach((point, index) => {
+            if(index !== 0) {
+                this.path2D.lineTo(point.x, point.y);
+            }
+        });
+        context.fillStyle = this.pickFillColor();
+        context.fill(this.path2D);
+        context.stroke(this.path2D);
+    }
+
+    pickFillColor() {
+        return this.intersectingShapes.length === 0? "#00FF00" : "#DC143C";          
+    }
+    move(offsets, mousePosition) {
+        this.points.forEach((point, index) => {
+            point.x = mousePosition.x + offsets[index].x;
+            point.y = mousePosition.y + offsets[index].y;
+        })
+    }
+}
+export const shapes = [
```

Comment:
extract points into a config file and create a specific service that would handle shapes (for example ShapesHandler or similar).